dojo/tools/kiuwan_sca/parser.py

```diff
@@ -0,0 +1,89 @@
+import json
+import hashlib
+
+from dojo.models import Finding
+
+__author__ = "mwager"
+
+
+class KiuwanSCAParser:
+    SEVERITY = {
+        "-": "Low",
+        "LOW": "Low",
+        "MEDIUM": "Medium",
+        "HIGH": "High",
+        "CRITICAL": "Critical",
+        "Low": "Low",
+        "Medium": "Medium",
+        "High": "High",
+        "Critical": "Critical"
+    }
+
+    def get_scan_types(self):
+        return ["Kiuwan SCA Scan"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return scan_type
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Import Kiuwan Insights Scan in JSON format. Export as JSON using Kiuwan REST API."
+
+    def get_findings(self, filename, test):
+        data = json.load(filename)
+        dupes = dict()
+
+        for row in data:
+            # if a finding was "muted" in the Kiuwan UI, we ignore it (e.g. marked as false positive)
+            if row["muted"] is True:
+                continue
+
+            finding = Finding(test=test)
+            finding.unique_id_from_tool = row["id"]
+            finding.cve = row["cve"]
+            finding.description = row["description"]
+            finding.severity = self.SEVERITY[row["securityRisk"]]
+
+            if "components" in row and len(row["components"]) > 0:
+                finding.component_name = row["components"][0]["artifact"]
+                finding.component_version = row["components"][0]["version"]
+                finding.title = finding.component_name + " v" + str(finding.component_version)
+
+            if not finding.title:
+                finding.title = row["cve"]
+
+            if "cwe" in row:
+                try:
+                    finding.cwe = int(row["cwe"].replace("CWE-", ""))
+                except Exception:
+                    pass
+
+            if "epss_score" in row:
+                finding.epss_score = row["epss_score"]
+            if "epss_percentile" in row:
+                finding.epss_percentile = row["epss_percentile"]
+
+            if "cVSSv3BaseScore" in row:
+                finding.cvssv3_score = float(row["cVSSv3BaseScore"])
+
+            finding.references = "See Kiuwan Web UI"
+            finding.mitigation = "See Kiuwan Web UI"
+            finding.static_finding = True
+
+            key = hashlib.sha256(
```

Comment:
Hi @mwager, the PR looks quite good.

I don't know if you need the key at all if you choose DEDUPE_ALGO_HASH_CODE.

Reply:
Thx! As I understood, this is only for deduping per import? i.e. removing dupes in *one* import file, therefore preventing further deduplication later? Maybe I got this wrong then?

![grafik](https://github.com/DefectDojo/django-DefectDojo/assets/999748/6fedda26-7c34-4af0-9de1-e6680178f98c)

But I can remove it if this is not needed. We tested it with lots of prod data and it does what we expect :)

Reply:
Let's wait for the reviews of the maintainers. Ah, good point, maybe you are right, but why would you have duplicates in one file?